index.ts

```diff
@@ -289,34 +325,27 @@ export function reverse ({
  * @param {string} [$0.url]                     optional URL to override ESRI reverseGeocode endpoint
  * @return {Promise}                            A Promise that'll get resolved with search result
  */
-export function search ({
+export function search({
+  boundary,
   clientId,
   clientSecret,
-  boundary,
   focusPoint,
   forStorage = false,
   magicKey,
   size = 10,
   text,
   url
-}: BaseQuery & {
-  boundary?: Boundary,
-  focusPoint?: any,
-  forStorage?: boolean,
-  magicKey?: string,
-  size?: number,
-  text: string,
-}): Promise<BaseResponse> {
-  const geocoder = getGeocoder(clientId, clientSecret, url)
-  const options = {}
+}: BaseQuery & Options): Promise<BaseResponse> {
```

Comment:
Two things:

1. Can you in-line the applicable keys within the `Options` type since not all of those values would apply as keys in the input argument for this function call?
2. Can you also please extend the `BaseResponse` type with the `query` key/value since the query is eventually returned?